src/panels/lovelace/components/hui-image.ts

```diff
@@ -156,46 +190,88 @@ export class HuiImage extends LitElement {
                 @load=${this._onImageLoad}
                 style=${styleMap({
                   filter,
-                  display: this._loadError ? "none" : "block",
+                  display:
+                    this._loadState === LoadState.Loaded ? "block" : "none",
                 })}
               />
             `}
-        ${this._loadError
+        ${this._loadState === LoadState.Error
           ? html`<div
               id="brokenImage"
               style=${styleMap({
                 height: `${this._lastImageHeight || "100"}px`,
               })}
             ></div>`
+          : imageSrc === undefined || this._loadState === LoadState.Loading
+          ? html`<div
+              style=${styleMap({
+                height: `${this._lastImageHeight || "100"}px`,
+              })}
+            >
+              <ha-circular-progress
+                class="render-spinner"
+                active
+                size="small"
+              ></ha-circular-progress>
+            </div>`
           : ""}
       </div>
     `;
   }
 
+  protected _shouldStartCameraUpdates(oldHass?: HomeAssistant): boolean {
+    return !!(
+      (!oldHass || oldHass.connected !== this.hass!.connected) &&
+      this.hass!.connected &&
+      this.cameraView !== "live"
+    );
+  }
+
   protected updated(changedProps: PropertyValues): void {
     if (changedProps.has("hass")) {
       const oldHass = changedProps.get("hass") as HomeAssistant | undefined;
-      if (!oldHass || oldHass.connected !== this.hass!.connected) {
-        if (this.hass!.connected && this.cameraView !== "live") {
-          this._updateCameraImageSrc();
-          this._startUpdateCameraInterval();
-        } else if (!this.hass!.connected) {
-          this._stopUpdateCameraInterval();
-          this._cameraImageSrc = undefined;
-          this._loadError = true;
-        }
+
+      if (this._shouldStartCameraUpdates(oldHass)) {
+        this._startIntersectionObserverOrUpdates();
+      } else if (!this.hass!.connected) {
+        this._stopUpdateCameraInterval();
+        // We used to set load error when stopping
+        // but that resulted in every image being broken
+        // when a phone is locked and unlocked
       }
     } else if (changedProps.has("cameraImage") && this.cameraView !== "live") {
-      this._updateCameraImageSrc();
+      this._startIntersectionObserverOrUpdates();
```

Comment:
We can combine this with the if on line 234?

Reply:
cleaned up when I moved it into willUpdate